src/lib/mlang.js

```diff
@@ -150,108 +232,259 @@ foam.CLASS({
   methods: [
     function f(o) {
-      return this.arg1.f(o) > this.arg2.f(o);
+      return foam.util.compare(this.arg1.f(o), this.arg2.f(o)) <= 0;
     }
   ]
 });
 
+/** Binary expression for "strictly greater than". */
+foam.CLASS({
+  package: 'foam.mlang.predicate',
+  name: 'Gt',
+  extends: 'foam.mlang.predicate.Binary',
+  methods: [
+    function f(o) {
+      return foam.util.compare(this.arg1.f(o), this.arg2.f(o)) > 0;
+    }
+  ]
+});
+
+/** Binary expression for "greater than or equal to". */
+foam.CLASS({
+  package: 'foam.mlang.predicate',
+  name: 'Gte',
+  extends: 'foam.mlang.predicate.Binary',
+  methods: [
+    function f(o) {
+      return foam.util.compare(this.arg1.f(o), this.arg2.f(o)) >= 0;
+    }
+  ]
+});
+
+
+/** Unary expression that checks the given property is well-defined. */
+foam.CLASS({
+  package: 'foam.mlang.predicate',
+  name: 'Has',
+  extends: 'foam.mlang.predicate.Unary',
+
+  methods: [
+    function f(obj) {
+      var value = this.arg1.f(obj);
+      var notHas = value === undefined || value === null || value === '' ||
+          (Array.isArray(value) && value.length === 0);
+      return !notHas;
+    }
+  ]
+});
+
+/** Unary expression that expects a boolean value and inverts it. */
+foam.CLASS({
+  package: 'foam.mlang.predicate',
+  name: 'Not',
+  extends: 'foam.mlang.predicate.Unary',
+
+  methods: [
+    function f(obj) {
+      return ! this.arg1.f(obj);
+    }
+  ]
+});
+
+
 foam.CLASS({
   package: 'foam.mlang',
   name: 'Expressions',
   requires: [
-    'foam.mlang.CountExpr',
-    'foam.mlang.ConstantExpr',
-    'foam.mlang.ContainsExpr',
-    'foam.mlang.OrExpr',
-    'foam.mlang.AndExpr',
-    'foam.mlang.InExpr',
-    'foam.mlang.EqExpr',
-    'foam.mlang.GtExpr'
+    'foam.mlang.predicate.And',
+    'foam.mlang.predicate.Constant',
+    'foam.mlang.predicate.Contains',
+    'foam.mlang.predicate.ContainsIC',
+    'foam.mlang.predicate.Eq',
+    'foam.mlang.predicate.Gt',
+    'foam.mlang.predicate.Gte',
+    'foam.mlang.predicate.Has',
+    'foam.mlang.predicate.In',
+    'foam.mlang.predicate.Lt',
+    'foam.mlang.predicate.Lte',
+    'foam.mlang.predicate.Neq',
+    'foam.mlang.predicate.Not',
+    'foam.mlang.predicate.Or',
+    'foam.mlang.sink.Count',
   ],
   methods: [
-    function _nary_(name, args) { return this[name].create({ args: foam.array.argsToArray(args) }); },
-    function _binary_(name, arg1, arg2) { return this[name].create({ arg1: arg1, arg2: arg2 }); },
-    function OR() { return this._nary_("OrExpr", arguments); },
-    function AND() { return this._nary_("AndExpr", arguments); },
-    function IN(a, b) { return this._binary_("InExpr", a, b); },
-    function CONTAINS(a, b) { return this._binary_("ContainsExpr", a, b); },
-    function EQ(a, b) { return this._binary_("EqExpr", a, b); },
-    function GT(a, b) { return this._binary_("GtExpr", a, b); }
+    function _nary_(name, args) {
+      return this[name].create({ args: foam.array.argsToArray(args) });
```

Comment:
FYI, it is faster to do:
var r = this[name].create();
r.args = foam.array.argsToArray(args);
return r;
But since the cost of creating an mLang is orders of magnitude less than using it, it doesn't matter.